pyartcd/pyartcd/util.py

```diff
@@ -267,3 +275,191 @@ async def is_build_permitted(version: str, data_path: str = constants.OCP_BUILD_
 
     # Fallback to default
     return True
+
+
+def getReleaseControllerArch(releaseStreamName):
+    arch = 'amd64'
+    streamNameComponents = releaseStreamName.split('-')
+    for goArch in goArches:
+        if goArch in streamNameComponents:
+            arch = goArch
+    return arch
+
+
+def getReleaseControllerURL(releaseStreamName):
+    arch = getReleaseControllerArch(releaseStreamName)
+    return f"https://{arch}.ocp.releases.ci.openshift.org"
+
+
+def stagePublishClient(working_dir, from_release_tag, release_name, arch, client_type):
```

Comment:
This function is overly complicated. Please consider splitting it to make it clearer and more maintainable